pkg/core/resources/apis/mesh/validators.go

```diff
@@ -348,10 +348,13 @@ func ValidateTargetRef(
 		}
 		err.Add(disallowedField("mesh", ref.Mesh, ref.Kind))
 		err.Add(disallowedField("tags", ref.Tags, ref.Kind))
-	case common_api.MeshSubset:
+	case common_api.MeshSubset, common_api.MeshGatewaysSubset:
 		err.Add(disallowedField("name", ref.Name, ref.Kind))
 		err.Add(disallowedField("mesh", ref.Mesh, ref.Kind))
 		err.Add(ValidateTags(validators.RootedAt("tags"), ref.Tags, ValidateTagsOpts{}))
+		if ref.Kind == common_api.MeshGatewaysSubset && len(ref.Tags) > 0 && !opts.GatewayListenerTagsAllowed {
+			err.Add(disallowedField("tags", ref.Tags, ref.Kind))
```

Comment:
Hmm this may be a bit tricky. What if you want to select a strict subset of mesh gateways, but all listeners of each of them?

Reply:
probably we should ensure that MeshGatewaysSubset can only select _all_ listeners of a subset of gateways. So we would ignore listener tags when matching

Reply:
but we can fix that in a later PR